jar-submission-test/src/main/java/com/hazelcast/jet/tests/jarsubmission/JarSubmissionTest.java

```diff
@@ -0,0 +1,310 @@
+                exceptions[3] = t;
+            }
+        });
+        executorService.shutdown();
+        executorService.awaitTermination((long) (durationInMillis + DELAY_AFTER_TEST_FINISHED), MILLISECONDS);
+
+        if (exceptions[0] != null) {
+            logger.severe("Exception in small batch jar.", exceptions[0]);
+        }
+        if (exceptions[1] != null) {
+            logger.severe("Exception in large batch jar.", exceptions[1]);
+        }
+        if (exceptions[2] != null) {
+            logger.severe("Exception in small stream jar.", exceptions[2]);
+        }
+        if (exceptions[3] != null) {
+            logger.severe("Exception in large stream jar.", exceptions[3]);
+        }
+
+        for (Throwable exception : exceptions) {
+            if (exception != null) {
+                throw exception;
+            }
+        }
+    }
+
+    @Override
+    protected void teardown(Throwable t) throws Exception {
+    }
+
+    protected void testBatchSmallJar(HazelcastInstance client) throws IOException {
+        testBatchJar(client, smallJarPath, "JarSubmissionTestBatchSmall", pauseBetweenSmallJarJobs,
+                SMALL_JAR_LOG_JOB_COUNT_THRESHOLD);
+    }
+
+    protected void testBatchLargeJar(HazelcastInstance client) throws IOException {
+        testBatchJar(client, largeJarPath, "JarSubmissionTestBatchLarge", pauseBetweenLargeJarJobs,
+                LARGE_JAR_LOG_JOB_COUNT_THRESHOLD);
+    }
+
+    protected void testStreamSmallJar(HazelcastInstance client) throws IOException {
+        testStreamJar(client, smallJarPath, "JarSubmissionTestStreamSmall", pauseBetweenSmallJarJobs,
+                SMALL_JAR_LOG_JOB_COUNT_THRESHOLD);
+    }
+
+    protected void testStreamLargeJar(HazelcastInstance client) throws IOException {
+        testStreamJar(client, largeJarPath, "JarSubmissionTestStreamLarge", pauseBetweenLargeJarJobs,
+                LARGE_JAR_LOG_JOB_COUNT_THRESHOLD);
+    }
+
+    protected void testBatchJar(HazelcastInstance client, String jarPath, String prefix, int sleep, int logThreshold) {
+        Path jarFile = Paths.get(jarPath);
+        assertTrue("testing file for " + prefix + " does not exist", Files.exists(jarFile));
+
+        JetService jet = client.getJet();
+
+        long begin = System.currentTimeMillis();
+        long jobCount = 0;
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            ArrayList<String> params = new ArrayList<>();
+            params.add(Long.toString(jobCount));
+            params.add(prefix);
+            // true means batch job
+            params.add("true");
+            SubmitJobParameters jobParams = new SubmitJobParameters()
+                    .setJobParameters(params)
+                    .setJobName(prefix + jobCount)
+                    .setJarPath(jarFile);
+            // set non-existing main class for 10% of jobs
+            if (jobCount % 10 == 5) {
```

Comment:
Maybe extract the variable to avoid duplications with module operation
```suggestion
            boolean jobWithNonExistingClass = jobCount % 10 == 5
            if (jobWithNonExistingClass) {
```